src/banded/dot.jl

```diff
@@ -0,0 +1,43 @@
+function _dot(x::AbstractVector, A::BandedMatrix, y::AbstractVector)
```

Comment:
What's the point of `_dot`? Just implement as `dot` below?

Reply:
The point is that the specialized method for `A.data isa FillArrays.AbstractFill` only works for a single diagonal, so if there is more than one diagonal, we dispatch to this generic implementation, which is also used for generic types of `A.data`.